library/x509_invasive.h

```diff
@@ -0,0 +1,45 @@
+/**
+ * \file x509_invasive.h
+ *
+ * \brief x509 module: interfaces for invasive testing only.
+ *
+ * The interfaces in this file are intended for testing purposes only.
+ * They SHOULD NOT be made available in library integrations except when
+ * building the library for testing.
+ */
+/*
+ *  Copyright The Mbed TLS Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+#ifndef MBEDTLS_X509_INVASIVE_H
+#define MBEDTLS_X509_INVASIVE_H
+
+#if defined(MBEDTLS_TEST_HOOKS)
+
+/*
+ * parse ipv4 address from canonical string form into bytes.
```

Comment:
I would suggest having doxygen syle comments here, describing the parameters and return values. e.g 
```
\param  h       dottted-deciamal ipv4 address, e.g. "192.168.12.88". Must be NUL-terminated
\param  hlen    length of the string at h.
\param  addr    pointer to a char array...

\return  0 if the string was parsed and the address bytes are stored in addr, -1 otherwise
```
I reckon if you have the parameter descriptions here, you probably do not require them in `x509_crt.c`, but if you intend to keep them there, I would also use doxygen style comments.